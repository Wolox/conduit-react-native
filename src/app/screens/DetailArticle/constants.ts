import { darkGray, green } from '@constants/colors';
import { SPACING } from '@constants/dimentions';
import { UserResponse } from '@interfaces/authInterfaces';
import { MixedStyleDeclaration } from 'react-native-render-html';

export function MOCK_DATA({ user: { username } }: UserResponse, comment: string) {
  return {
    author: {
      bio: 'Bio de prebas',
      following: false,
      image:
        'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8dXNlcnxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&w=1000&q=80',
      username
    },
    body: comment,
    createdAt: '2021-08-06T18:16:15.699Z',
    updatedAt: '2021-08-06T18:16:15.699Z',
    id: Number(Math.random().toFixed(2))
  };
}
enum COMMON_STYLES {
  COMMON_TITLE_FONT = 14,
  PARAGRAPH_SIZE = 12,
  NONE = 'none',
  NORMAL = 'normal'
}
const TITLE_STYLES: MixedStyleDeclaration = {
  fontSize: COMMON_STYLES.COMMON_TITLE_FONT,
  padding: SPACING.NONE,
  marginVertical: SPACING.XXSMALL
};
export const tagsStyles: Readonly<Record<string, MixedStyleDeclaration>> = {
  body: {
    whiteSpace: COMMON_STYLES.NORMAL,
    color: darkGray
  },
  h1: {
    ...TITLE_STYLES
  },
  h2: {
    ...TITLE_STYLES
  },
  p: {
    fontSize: COMMON_STYLES.PARAGRAPH_SIZE,
    padding: SPACING.NONE,
    marginVertical: SPACING.XXSMALL
  },
  a: {
    color: green,
    textDecorationLine: COMMON_STYLES.NONE
  },
  ol: {
    margin: SPACING.NONE,
    padding: SPACING.NONE
  },
  li: {
    paddingBottom: SPACING.XXSMALL
  }
};
