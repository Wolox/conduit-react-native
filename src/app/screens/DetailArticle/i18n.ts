import i18next from 'i18next';

i18next.addResources('en', 'DETAIL_ARTICLE', {
  NO_COMMENTS_TEXT: ' No comments yet, Do the first!',
  PLACE_HOLDER_INPUT: 'Insert your comment...',
  NO_AUTHENTICATED: 'Register to interact with other users',
  MIN_LENGTH_COMMENT: 'A comment requires at least 5 characters'
});
