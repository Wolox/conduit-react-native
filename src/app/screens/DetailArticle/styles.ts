import { StyleSheet } from 'react-native';
import { darkGray, lightGray, white, red, gray, green } from '@constants/colors';
import { SPACING, BORDER_WIDTH } from '@constants/dimentions';
import { BOTTOM_SEPARATOR_LINE, commonBoxShadow } from '@constants/commonStyles';
import { moderateScale } from '@utils/scalingUtils';
import fonts from '@config/fonts';

const IMAGE_USER_SIZE = moderateScale(80);
const INTERACION_ICON_SIZE = moderateScale(30);

export default StyleSheet.create({
  container: {
    flex: 1,
    position: 'relative',
    alignItems: 'center',
    padding: SPACING.MEDIUM,
    paddingTop: SPACING.XXBIG * SPACING.XXXSMALL
  },
  containerActions: {
    width: '100%',
    height: INTERACION_ICON_SIZE,
    position: 'absolute',
    top: SPACING.MEDIUM,
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  containerDetail: {
    ...commonBoxShadow,
    position: 'relative',
    padding: SPACING.MEDIUM,
    borderRadius: SPACING.MEDIUM,
    backgroundColor: white,
    paddingTop: SPACING.XXBIG
  },
  containerUser: {
    marginBottom: SPACING.MEDIUM,
    justifyContent: 'center',
    alignItems: 'center'
  },
  image: {
    position: 'absolute',
    top: -SPACING.XXBIG * SPACING.XXXSMALL,
    width: IMAGE_USER_SIZE,
    height: IMAGE_USER_SIZE,
    backgroundColor: darkGray,
    borderRadius: SPACING.XXBIG
  },
  interactionButtons: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: SPACING.SMALL,
    height: INTERACION_ICON_SIZE,
    width: '100%'
  },
  interactionButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: SPACING.XSMALL
  },
  interactionButtonImage: {
    width: INTERACION_ICON_SIZE,
    height: INTERACION_ICON_SIZE,
    marginRight: SPACING.XSMALL,
    tintColor: gray
  },
  separator: {
    ...BOTTOM_SEPARATOR_LINE,
    marginVertical: SPACING.SMALL
  },
  tagContainer: {
    flexDirection: 'row'
  },
  tag: {
    ...fonts.labelFont,
    borderColor: lightGray,
    borderWidth: BORDER_WIDTH.BOLD,
    padding: SPACING.XXXSMALL,
    marginHorizontal: SPACING.XXXSMALL
  },
  redTint: {
    tintColor: red
  },
  greenTint: {
    tintColor: green
  }
});
