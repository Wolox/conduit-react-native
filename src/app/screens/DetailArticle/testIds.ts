export default {
  favoriteButton: 'favorite-button',
  followButton: 'follow-button'
};
