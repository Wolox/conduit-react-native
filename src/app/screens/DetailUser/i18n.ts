import i18next from 'i18next';

i18next.addResources('en', 'DETAIL_USER', {
  DESCRIPTION: 'Description',
  DESCRIPTION_BLANK: 'No description yet',
  MY_ARTICLES: 'My Articles',
  FAVORITED_ARTICLES: 'Favorited Articles',
  USER_FOLLOW: 'Following',
  USER_UNFOLLOW: 'Follow'
});
