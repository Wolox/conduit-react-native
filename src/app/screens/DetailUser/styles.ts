import { StyleSheet } from 'react-native';
import { BOTTOM_SEPARATOR_LINE } from '@constants/commonStyles';
import { SPACING } from '@constants/dimentions';
import { grayTransparent, green } from '@constants/colors';

const styles = StyleSheet.create({
  containerBio: {
    width: '100%',
    marginVertical: 12,
    padding: 5
  },
  containerArticles: {
    flex: 1
  },
  description: {
    width: '100%',
    marginTop: 10
  },
  separator: {
    ...BOTTOM_SEPARATOR_LINE,
    width: '100%',
    alignItems: 'center'
  },
  titleEmptyArticles: {
    marginTop: SPACING.MEDIUM
  },
  followButtonContainer: {
    alignItems: 'flex-end',
    marginHorizontal: SPACING.SMALL
  },
  interactionButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    width: '30%',
    padding: 10,
    borderRadius: 50
  },
  following: {
    backgroundColor: green
  },
  unfollowing: {
    backgroundColor: grayTransparent
  },
  greenTint: {
    tintColor: green
  }
});
export default styles;
