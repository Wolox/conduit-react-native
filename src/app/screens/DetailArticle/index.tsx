import React, { useState, useCallback } from 'react';
import { View, Image, TouchableOpacity, Text } from 'react-native';
import { useSelector } from 'react-redux';
import CustomText from '@components/CustomText';
import useNavigation from '@components/AppNavigator/helper';
import { State } from '@interfaces/reduxInterfaces';
import { ArticleInParams } from '@interfaces/articlesInterface';
import { formatDate } from '@utils/dateUtils';
import icAddInactive from '@assets/TabBar/icAddpostInactive.png';
import icAddActive from '@assets/TabBar/icAddpostActive.png';
import icFavouriteInactive from '@assets/TabBar/icFavoriteInactive.png';
import icFavouriteActive from '@assets/TabBar/icFavoriteActive.png';
import icDefaultArticleImage from '@assets/icons/icDefaultArticleImage.jpg';
import icDelete from '@assets/icons/icDelete.png';
import icEdit from '@assets/icons/icEdit.png';
import Routes from '@constants/routes';

import styles from './styles';
import testIds from './testIds';

interface Props extends ArticleInParams {}

function DetailArticle({ route }: Props) {
  const {
    title,
    description,
    updatedAt,
    favoritesCount,
    author: { image, username, following },
    tagList
  } = route?.params?.article;
  const currentUser = useSelector((state: State) => state.auth.currentUser);
  const navigation = useNavigation();
  const [favoriteCount, setFavoriteCount] = useState(favoritesCount || 0);
  const [isFollow, setIsFollow] = useState(following);
  const handleToggleFavorite = () => {
    if (favoriteCount > favoritesCount) setFavoriteCount(favoriteCount - 1);
    else setFavoriteCount(favoriteCount + 1);
  };
  const handleDeleteArticle = () => console.log('delete article');

  const handleEditArticle = useCallback(
    () =>
      navigation?.navigate(Routes.EditArticle, {
        article: { ...route?.params?.article, isEditArticle: true }
      }),
    [navigation, route]
  );

  return (
    <View style={styles.container}>
      {currentUser && (
        <View style={styles.containerActions}>
          <TouchableOpacity
            testID={testIds.editButton}
            style={styles.interactionButton}
            onPress={handleEditArticle}>
            <Image
              style={[styles.interactionButtonImage, styles.greenTint]}
              source={icEdit}
              resizeMode="contain"
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={testIds.deleteButton}
            style={styles.interactionButton}
            onPress={handleDeleteArticle}>
            <Image
              style={[styles.interactionButtonImage, styles.redTint]}
              source={icDelete}
              resizeMode="contain"
            />
          </TouchableOpacity>
        </View>
      )}
      <View style={styles.containerDetail}>
        <View style={styles.containerUser}>
          <Image source={image ? { uri: image } : icDefaultArticleImage} style={styles.image} />
          <View>
            <CustomText center green>
              {username}
            </CustomText>
            <CustomText center label>
              {formatDate(updatedAt)}
            </CustomText>
          </View>
        </View>
        {!!tagList.length && (
          <View style={styles.tagContainer}>
            {tagList.map((tag: string, index: number) => (
              <Text style={styles.tag} key={`key-${tag}-${index}`}>
                {tag}
              </Text>
            ))}
          </View>
        )}
        <View style={styles.separator} />
        <CustomText>{title}</CustomText>
        <CustomText label>{description}</CustomText>
        <View style={styles.interactionButtons}>
          <TouchableOpacity
            testID={testIds.followButton}
            style={styles.interactionButton}
            onPress={() => setIsFollow(!isFollow)}>
            <Image
              style={styles.interactionButtonImage}
              source={isFollow ? icAddActive : icAddInactive}
              resizeMode="contain"
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={testIds.favoriteButton}
            style={styles.interactionButton}
            onPress={handleToggleFavorite}>
            <Image
              style={styles.interactionButtonImage}
              source={favoriteCount > favoritesCount ? icFavouriteActive : icFavouriteInactive}
              resizeMode="contain"
            />
            {!!favoriteCount && (
              <CustomText
                gray
                xsmall
                green={favoriteCount > favoritesCount}>{`(${favoriteCount})`}</CustomText>
            )}
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

export default DetailArticle;
