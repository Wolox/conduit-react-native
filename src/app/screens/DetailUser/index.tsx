import React, { useCallback, useEffect, useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import i18next from 'i18next';
import ProfileActions from '@redux/profile/actions';
import WithHeader from '@components/WithHeader';
import CustomText from '@components/CustomText';
import TabList from '@components/TabList';
import { getAvatar } from '@constants/iconsConstants';
import icDefaultArticleImage from '@assets/icons/icDefaultArticleImage.jpg';
import { ArticlesState, State } from '@interfaces/reduxInterfaces';
import { Article, Author } from '@interfaces/articlesInterface';
import ArticlesActions from '@redux/articles/actions';

import RenderContent from './components/RenderContent';
import './i18n';
import styles from './styles';
import testIds from './testIds';

type Props = {
  route: {
    params: {
      user: string;
    };
  };
};

export default function DetailUser({
  route: {
    params: { user }
  }
}: Props) {
  const dispatch = useDispatch();
  const [currentTab, setCurrentTab] = useState(0);
  const currentUser = useSelector((state: State) => state?.auth?.currentUser?.user?.username);
  const articlesByAuthor = useSelector<State, Article[]>(
    state => state.articles.articlesAuthor.articles || []
  );
  const favoritedByAuthor = useSelector<State, Article[]>(
    state => state.articles.favoritedAuthor.articles || []
  );
  const { articlesAuthorLoading, favoritedAuthorLoading } = useSelector<State, ArticlesState>(
    state => state.articles || []
  );
  const userProfile = useSelector<State, Author>(state => state.profile?.profileUser?.profile);
  const { bio, image, username, following } = userProfile || {};
  useEffect(() => {
    dispatch(ProfileActions.getProfileUser(user));
  }, [dispatch, user]);
  const handlePressTab = (index: number) => setCurrentTab(index);

  const getArticlesByAuthor = useCallback(async () => {
    await dispatch(ArticlesActions.getArticleByAuthor(user));
  }, [dispatch, user]);

  const getFavoritedByAuthor = useCallback(async () => {
    await dispatch(ArticlesActions.getFavoritedByAuthor(user));
  }, [dispatch, user]);

  useEffect(() => {
    if (currentTab === 1) {
      getFavoritedByAuthor();
    } else {
      getArticlesByAuthor();
    }
  }, [currentTab, getArticlesByAuthor, getFavoritedByAuthor]);

  const [isFollow, setIsFollow] = useState(following);

  const handleFollowUser = useCallback(() => {
    dispatch(ProfileActions.followUser(username, isFollow));
    setIsFollow(!isFollow);
  }, [dispatch, isFollow, username]);

  return (
    <WithHeader
      title={i18next.t(`app:${username}`)}
      withAvatar
      avatar={image ? getAvatar(image) : icDefaultArticleImage}>
      <View style={styles.containerBio}>
        {currentUser !== username && (
          <View style={styles.followButtonContainer}>
            <TouchableOpacity
              testID={testIds.followUserButton}
              style={[
                styles.interactionButton,
                isFollow && styles.following,
                !isFollow && styles.unfollowing
              ]}
              onPress={handleFollowUser}>
              <CustomText white={isFollow} bold center>
                {isFollow ? i18next.t('DETAIL_USER:USER_FOLLOW') : i18next.t('DETAIL_USER:USER_UNFOLLOW')}
              </CustomText>
            </TouchableOpacity>
          </View>
        )}
        <CustomText green medium center>
          {i18next.t('DETAIL_USER:DESCRIPTION')}
        </CustomText>
        <CustomText center style={styles.description}>
          {bio?.length ? bio : i18next.t('DETAIL_USER:DESCRIPTION_BLANK')}
        </CustomText>
      </View>
      <TabList
        tabs={[i18next.t('DETAIL_USER:MY_ARTICLES'), i18next.t('DETAIL_USER:FAVORITED_ARTICLES')]}
        onPressTab={handlePressTab}
      />
      {currentTab === 0
        ? articlesByAuthor && <RenderContent loading={articlesAuthorLoading} data={articlesByAuthor} />
        : favoritedByAuthor && <RenderContent loading={favoritedAuthorLoading} data={favoritedByAuthor} />}
    </WithHeader>
  );
}
