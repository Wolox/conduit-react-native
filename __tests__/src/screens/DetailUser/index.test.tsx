import React from 'react';
import renderer from 'react-test-renderer';
import DetailArticle from '@screens/DetailArticle';
import userIcon from '@assets/Profile/icUser.png';
import MockedNavigator from '@mocks/MockedNavigator';
import DetailUser from '@screens/DetailUser';

jest.mock('react-redux', () => {
  return {
    useSelector: jest.fn().mockImplementation(() => ({ comments: [], currentUser: '' })),
    useDispatch: () => jest.fn()
  };
});
const props = {
  route: {
    params: {
      user: 'hola'
    }
  }
};

describe('DetailArticle', () => {
  test('DetailArticle Snapshot', () => {
    const RenderCustom = {
      renderDetail: () => <DetailUser {...props} />,
      renderCompleteDetail: () => <MockedNavigator component={RenderCustom.renderDetail} />
    };
    const tree = renderer.create(RenderCustom.renderCompleteDetail()).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
