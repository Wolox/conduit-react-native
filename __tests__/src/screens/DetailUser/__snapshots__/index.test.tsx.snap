// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailArticle DetailArticle Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#5CB85C",
                          "width": "100%",
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "minHeight": 168.14285714285714,
                              "paddingHorizontal": 16,
                              "paddingVertical": 20,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 2,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "color": "#FFF",
                                      "fontFamily": "Nunito",
                                      "fontSize": 31.428571428571427,
                                      "fontStyle": "normal",
                                      "fontWeight": "700",
                                    },
                                  ],
                                  Object {
                                    "marginBottom": 5,
                                  },
                                ]
                              }
                            >
                              app:undefined
                            </Text>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "height": 14,
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#5CB85C",
                                "flex": 1,
                              },
                            ]
                          }
                        />
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FFF",
                              "borderTopLeftRadius": 14,
                              "borderTopRightRadius": 14,
                              "height": "100%",
                              "overflow": "hidden",
                              "position": "absolute",
                              "top": 0,
                              "width": "100%",
                            },
                            Object {
                              "paddingTop": 97.14285714285714,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginVertical": 12,
                              "padding": 5,
                              "width": "100%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "flex-end",
                                "marginHorizontal": 12,
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#00000033",
                                  "borderRadius": 50,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "opacity": 1,
                                  "padding": 10,
                                  "width": "30%",
                                }
                              }
                              testID="follow-user-button"
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "color": "#000",
                                        "fontFamily": "Nunito",
                                        "fontSize": 25.142857142857142,
                                        "fontStyle": "normal",
                                        "fontWeight": "700",
                                      },
                                      Object {
                                        "textAlign": "center",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                DETAIL_USER:USER_UNFOLLOW
                              </Text>
                            </View>
                          </View>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [
                                  Object {
                                    "textAlign": "center",
                                  },
                                  Object {
                                    "color": "#5CB85C",
                                  },
                                  Object {
                                    "fontSize": 25.142857142857142,
                                  },
                                ],
                                undefined,
                              ]
                            }
                          >
                            DETAIL_USER:DESCRIPTION
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [
                                  Object {
                                    "textAlign": "center",
                                  },
                                ],
                                Object {
                                  "marginTop": 10,
                                  "width": "100%",
                                },
                              ]
                            }
                          >
                            DETAIL_USER:DESCRIPTION_BLANK
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "paddingHorizontal": 16,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "width": "50%",
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": "#5CB85C",
                                  "borderBottomWidth": 3,
                                  "opacity": 1,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#5CB85C",
                                      },
                                    ],
                                    Object {
                                      "margin": 10,
                                    },
                                  ]
                                }
                              >
                                DETAIL_USER:MY_ARTICLES
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "width": "50%",
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": "#E5E5E5",
                                  "borderBottomWidth": 3,
                                  "opacity": 1,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "textAlign": "center",
                                      },
                                    ],
                                    Object {
                                      "margin": 10,
                                    },
                                  ]
                                }
                              >
                                DETAIL_USER:FAVORITED_ARTICLES
                              </Text>
                            </View>
                          </View>
                        </View>
                        <RCTScrollView
                          ItemSeparatorComponent={[Function]}
                          ListEmptyComponent={<RenderEmpty />}
                          data={Array []}
                          disableVirtualization={false}
                          getItem={[Function]}
                          getItemCount={[Function]}
                          horizontal={false}
                          initialNumToRender={10}
                          keyExtractor={[Function]}
                          maxToRenderPerBatch={10}
                          onContentSizeChange={[Function]}
                          onEndReachedThreshold={2}
                          onLayout={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          showsVerticalScrollIndicator={false}
                          stickyHeaderIndices={Array []}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                          updateCellsBatchingPeriod={50}
                          viewabilityConfigCallbackPairs={Array []}
                          windowSize={21}
                        >
                          <View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "textAlign": "center",
                                    },
                                  ],
                                  Object {
                                    "marginTop": 16,
                                  },
                                ]
                              }
                            >
                              HOME:EMPTY_ARTICLES
                            </Text>
                          </View>
                        </RCTScrollView>
                      </View>
                      <View
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "position": "absolute",
                            "top": -117.14285714285714,
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "process": [Function],
                            }
                          }
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#F1F1F1",
                                "borderColor": "#FFF",
                                "borderRadius": 107.14285714285714,
                                "borderWidth": 0.5,
                                "height": 214.28571428571428,
                                "width": 214.28571428571428,
                              },
                              false,
                            ]
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
